Engine/SubscriptionDataReader.cs

```diff
@@ -554,23 +558,29 @@ private SubscriptionStreamReader GetReader(string source)
                     // check if this is not a local uri then download it to the local cache
                     if (uri.IsAbsoluteUri && !uri.IsLoopback)
                     {
-                        try
+                        //Skip download if the file is already in cache
+                        if (!File.Exists(location))
```

Comment:
I think we may be able to accomplish the desired behavior without modifying the `SubscriptionDataConfig`.  Without knowing your exact use case, it seems like you're trying to download some data (possibly large?) only once, and then reference it from disk.  I think we may be able to put this logic in the `GetSource` method implemented by your custom data type.  The idea here would be that `GetSource` would check to see if your file exists, if not, then it would go and download it.  After the download completes, it can return the file path to the downloaded file.

I prefer a solution like this because it has the least impact on the system as a whole.  Does this solution fit your use case?